pdc_client/plugins/product_version.py

```diff
@@ -0,0 +1,146 @@
+
+import json
+
+from pdc_client.plugin_helpers import (PDCClientPlugin,
+                                       extract_arguments,
+                                       add_create_update_args)
+
+
+class ProductVersionPlugin(PDCClientPlugin):
+    def register(self):
+        self.set_command('product-version')
+
+        list_parser = self.add_action('list', help='list all product versions')
+        self.add_filter_arguments(list_parser)
+        list_parser.set_defaults(func=self.list_product_versions)
+
+        info_parser = self.add_action('info', help='display details of a product version')
+        info_parser.add_argument('product_version_id', metavar='PRODUCT_VERSION_ID')
+        info_parser.set_defaults(func=self.product_version_info)
+
+        update_parser = self.add_action('update', help='update an existing product version')
+        update_parser.add_argument('product_version_id', metavar='PRODUCT_VERSION_ID')
+        self.add_product_version_arguments(update_parser)
+        update_parser.set_defaults(func=self.product_version_update)
+
+        create_parser = self.add_action('create', help='create new product version')
+        self.add_product_version_arguments(create_parser, required=True)
+        create_parser.set_defaults(func=self.product_version_create)
+
+    def add_product_version_arguments(self, parser, required=False):
+        required_args = {
+            'name': {},
+            'product': {},
+            'version': {}}
+
+        optional_args = {
+            'short': {}}
+        add_create_update_args(parser, required_args, optional_args, required)
+
+        self.run_hook('product_parser_setup', parser)
+
+    def add_filter_arguments(self, parser):
+        active_group = parser.add_mutually_exclusive_group()
+        active_group.add_argument('--inactive', action='store_true', help='show only inactive products')
+        active_group.add_argument('--all', action='store_true', help='show both active and inactive products')
+
+        parser.add_argument('--name', help='show only products with matching name')
+        parser.add_argument('--version-id', help='show only products with matching product version ID')
+        parser.add_argument('--short', help='show only products with matching short name')
+        parser.add_argument('--version', help='show only products with matching version')
+
+    def list_product_versions(self, args):
+        filters = {}
+        if args.inactive:
+            filters['active'] = False
+        elif not args.all:
+            filters['active'] = True
+        if args.name:
+            filters['name'] = args.name
+        if args.version_id:
+            filters['product_version_id'] = args.version_id
+        if args.short:
+            filters['short'] = args.short
+        if args.version:
+            filters['version'] = args.version
+
+        product_versions = list(self.client.get_paged(self.client['product-versions']._, **filters))
```

Comment:
When running the patch with --page-size optional arg, can't return the results as expected. 
For example: 
$     ./bin/pdc -s dt -k --page-size 10  product-version list
It will return the results as one page,  the --page-size arg may be disabled.
Prefer to update the code: 
 ` product_versions = self.client.get_paged(self.client['product-versions']._, **filters)`
and update other codes as sync.